boards/posix/native_posix/doc/board.rst

```diff
@@ -458,8 +460,8 @@ The following peripherals are currently provided with this board:
     ``stdout``.
 
   - Feed any input from the native application ``stdin`` to a possible
-    running :ref:`Shell`. For more information refer to the section
-    `Shell support`_.
+    running legacy shell. For more information refer to the section
```

Comment:
Not the new shell, eh?  Will the new and legacy shells work together or are they mutually exclusive?

Reply:
> Not the new shell, eh? 

Not with that console driver. The new shell is not using ``CONSOLE`` , but hooks directly into the UART driver. So all the ``CONSOLE`` backends we had are of no use with the new shell.

> Will the new and legacy shells work together or are they mutually exclusive?

I guess the question should be for the new shell developers. But my understanding is that, today:
* The modules using the legacy shell are being ported to use the new shell. So the legacy shell will be purposeless quite soon (unless some odd app uses it).
* The new shell does not know or care about the old shell, and does not intend to avoid any conflict with it. If the old shell was driven thru UART they will certainly conflict if compiled together.
* In case of native_posix, I think you may still be able to compile the old shell in parallel with the new one by using this native console driver for the old shell, and the new UART driver for the new shell.

Reply:
AFAI legacy shell will be removed completely